src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -422,7 +371,44 @@ public static string UrlEncode(string value)
 
         public static byte[] UrlEncodeToBytes(byte[] value, int offset, int count)
         {
-            return UrlEncode(value, offset, count, true /* alwaysCreateNewReturnValue */);
+            if (!ValidateUrlEncodingParameters(value, offset, count))
+            {
+                return null;
+            }
+
+            int cSpaces = 0;
+            int cUnsafe = 0;
+
+            // count them first
+            for (int i = 0; i < count; i++)
+            {
+                char ch = (char)value[offset + i];
+
+                if (ch == ' ')
+                    cSpaces++;
+                else if (!IsUrlSafeChar(ch))
+                    cUnsafe++;
+            }
+
+            // nothing to expand?
+            if (cSpaces == 0 && cUnsafe == 0)
+            {
+                if (offset == 0 && value.Length == count)
```

Comment:
Did I introduce an extra branch? This would have occured anyway in the old method here https://github.com/dotnet/corefx/pull/7999/commits/65083bf69e4589c66c8b1d626886cb96b827c7e6#diff-831332bbb389db07350457a4d92095f8L293